cachi2/core/utils.py

```diff
@@ -197,6 +198,12 @@ def get_cache_dir() -> Path:
     return cache_dir.joinpath("cachi2")
 
 
-def first(predicate: Callable, iterable: Iterable, fallback: Any) -> Any:
+def first_for(predicate: Callable, iterable: Iterable, fallback: Any) -> Any:
     """Return the first match of predicate in iterable or fallback value."""
     return next((x for x in iterable if predicate(x)), fallback)
+
+
+def partition_by(predicate: Callable, iterable: Iterable) -> tuple[Iterable, Iterable]:
+    """Partition iterable in two by predicate."""
+    i1, i2 = tee(iterable)
+    return filterfalse(predicate, i1), filter(predicate, i2)
```

Comment:
similarly to `first`, feels like `partition_by` could be a standalone patch OR at least paired with `first` in a standalone patch without other logic around.